spec/controllers/products_controller_spec.rb

```diff
@@ -224,4 +224,62 @@ def mock_post_request(name, price, image)
       end
     end
   end
+
+  describe 'DELETE #destroy' do
+    context 'unauthenticated user' do
+      it 'redirects to sign up form' do
+        delete :destroy, params: { id: 5 }
+        expect(response).to redirect_to(new_user_session_path)
+      end
+    end
+
+    context 'authenticated user' do
+      let(:user) { create(:user) }
+      let!(:product_a) { create(:product, name: 'product_a', user: user) }
+      let!(:product_b) { create(:product, name: 'product_b', user: user) }
+      before do
+        mock_authentication
+        delete :destroy, params: { id: product_a.id }
+      end
+
+      it 'deletes correct product' do
+        expect(Product.find_by(id: product_a.id)).to be(nil)
+      end
+
+      it 'deletes ony one product' do
+        delete :destroy, params: { id: product_a.id }
+        expect(Product.all.count).to be(1)
```

Comment:
Lo que te voy a decir es equivalente en funcionamiento, pero tiendo a creer que es mejor eliminar un poco la dependencia de las creaciones anteriores. Me explico: si después creas un `:product_c` y solo borras 1, entonces este `.to be(1)` necesita cambiarse también.

Te recomendaría usar algo como `expect { acción }.to change(Modelo, :count).by(X)` o similar.